### ISSUE TITLE
Can't compile do construct without bind statement

### ISSUE THREAD

**Issue by kasperlanger** [puffnfresh/roy#59]
Here's an example of what I'm trying to do

```
let deferred = {
  return: \x -> $.when(x)
  bind: \x f -> x.done(f)
}

let v = (do deferred
  return "hello world"
)

(do deferred
  val <- v
  console.log val
)
```

I get the error `TypeError: Cannot set property 'rest' of undefined` in the second `do`

**Comment by puffnfresh** [comment]
Thanks, should be fixed in 9f1f5a83659719a8236b935e7a2e04907ed9b3c9.

**Comment by kasperlanger** [comment]
Thank you! Great project
